File: bug64_missing_edge_case_check/bug64_missing_edge_case_check/main1.cpp
#include <iostream>

int main() {
    int datasets[4][4] = {
        {0, 0, 5, 3},
        {7, 2, 0, 1},
        {0, 0, 0, 0},
        {4, 0, 0, 6}
    };

    unsigned long long acc = 0;

    for (int i = 0; i < 4; i++) {
        int f = 0;
        bool has_f = false;

        for (int j = 0; j < 4; j++) {
            if (datasets[i][j] != 0) {
                f = datasets[i][j];
                has_f = true;
                break;
            }
        }

        if (!has_f) {
            continue; // ✅ FIX
        }

        acc ^= (unsigned long long)f * (i + 1) * 0x9E3779B97F4A7C15ULL;
    }

    std::cout << std::hex << acc << std::endl; // ✅ Added output
    return 0;
}

File: bug64_missing_edge_case_check/bug64_missing_edge_case_check/main1.js
function main() {
    let datasets = [
        [0, 0, 5, 3],
        [7, 2, 0, 1],
        [0, 0, 0, 0],
        [4, 0, 0, 6]
    ];

    let acc = 0n;

    for (let i = 0; i < 4; i++) {
        let f = null;

        for (let j = 0; j < 4; j++) {
            if (datasets[i][j] !== 0) {
                f = datasets[i][j];
                break;
            }
        }

        if (f === null) {
            continue; // ✅ FIX
        }

        acc ^= BigInt(f) * BigInt(i + 1) * 0x9E3779B97F4A7C15n;
    }

    console.log(acc.toString(16)); // ✅ Added output
}
main();